### DISCUSSION TITLE
Shift+Return to leave current code block (i.e. remove indentation on new line)

### DISCUSSION THREAD

**Discussion by AlexVSharp** [godotengine/godot-proposals#5023] [Editor]
A simple quality of life suggestion. Someone [asked about this](https://godotengine.org/qa/52072/how-to-create-new-line-in-godot-text-editor-without-indent) a long time ago.

By default, the editor adds indentation (as required) once we press Return. This speeds up writing blocks of code.
However, to leave the current block, right now we always have to manually backspace after entering the new line.

I propose introducing a Shift+Return shortcut to do this automatically. i.e. Enter a new line with one (or more?) less indentation.
'Tis always one less key pressed. :+1:

**Comment by Calinou** [comment]
Note that there are already <kbd>Ctrl + Enter</kbd> and <kbd>Ctrl + Shift + Enter</kbd> shortcuts to go to the next/previous line without breaking in the middle of the line (even if your cursor is not at the end of the line).

<kbd>Shift + Enter</kbd> doesn't do anything different from pressing <kbd>Enter</kbd> in 3.5 (and it doesn't do anything in 4.0.alpha), so the proposed behavior sounds good.
